src/Tasks/Microsoft.Common.props

```
@@ -49,10 +49,10 @@ Copyright (C) Microsoft Corporation. All rights reserved.
       management system can write out multiple files but the order of the import is alphabetic because MSBuild sorts the list.
   -->
   <PropertyGroup>
-    <MSBuildProjectExtensionsPath>$([MSBuild]::EnsureTrailingSlash($([MSBuild]::ValueOrDefault('$(MSBuildProjectExtensionsPath)', '$(BaseIntermediateOutputPath)'))))</MSBuildProjectExtensionsPath>
+    <MSBuildProjectExtensionsPath>$([MSBuild]::EnsureTrailingSlash($([MSBuild]::ValueOrDefault('$(MSBuildProjectExtensionsPath)', '$(BuildDir)'))))</MSBuildProjectExtensionsPath>
```

This was expecting the `obj` folder previously. I'm confused what `BuildDir` is supposed to represent exactly. 

```
    BuildDir:
    Indicates the final output location for the project or solution.
    All the *OutpuPath properties should derive from this.
```

This suggests to me that it would be bin, so this could break whatever takes a dependency on MSBuildProjectExtensionsPath and expects the obj directory.

---

Ok, this is a mistake during rebase... silly me!!!

It was supposed to be...

```suggestion
    <MSBuildProjectExtensionsPath>$([MSBuild]::EnsureTrailingSlash($([MSBuild]::ValueOrDefault('$(MSBuildProjectExtensionsPath)', '$(BuildDir)ext'))))</MSBuildProjectExtensionsPath>
```

**Note**
Just like `bin` and `obj`, I've introduced `ext` folder to hold all the project restores, external tools hooking up with msbuild, etc... That way the outputs from these tools are kept separate, since, the pre-build events take a long time in the build process. This way, they are easy to include/copy/ignore in cache-enabled builds (e.g.: docker builds).